lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py

```diff
@@ -0,0 +1,264 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_devtestlab_facts
+version_added: "2.8"
+short_description: Get Azure DevTest Lab facts.
+description:
+    - Get facts of Azure DevTest Lab.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+    name:
+        description:
+            - The name of the lab.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+'''
+
+EXAMPLES = '''
+  - name: List instances of DevTest Lab by resource group
+    azure_rm_devtestlab_facts:
+      resource_group: testrg
+
+  - name: List instances of DevTest Lab in subscription
+    azure_rm_devtestlab_facts:
+
+  - name: Get instance of DevTest Lab
+    azure_rm_devtestlab_facts:
+      resource_group: testrg
+      name: testlab
+'''
+
+RETURN = '''
+labs:
+    description: A list of dictionaries containing facts for Lab.
+    returned: always
+    type: complex
+    contains:
+        id:
+            description:
+                - The identifier of the resource.
+            returned: always
+            type: str
+            sample: /subscriptions/{subscription-id}/resourcegroups/testrg/providers/microsoft.devtestlab/labs/testlab
+        resource_group:
+            description:
+                - The name of the resource.
+            returned: always
+            type: str
+            sample: testrg
+        name:
+            description:
+                - The name of the resource.
+            returned: always
+            type: str
+            sample: testlab
+        location:
+            description:
+                - The location of the resource.
+            returned: always
+            type: str
+            sample: eastus
+        storage_type:
+            description:
+                - Lab storage type.
+            returned: always
+            type: str
+            sample: standard
+        premium_data_disks:
+            description:
+                - Are premium data disks allowed.
+            returned: always
+            type: bool
+            sample: false
+        provisioning_state:
+            description:
+                - Lab provisioning state.
+            returned: always
+            type: str
+            sample: Succeeded
+        artifacts_storage_account:
+            description:
+                - Artifacts storage account ID.
+            returned: always
+            type: str
+            sample: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/testrg/providers/Microsoft.Storage/storageAccounts/testlab6346
```

Comment:
sample name consistent

Reply:
"testlab6346" is a storage account name and it's generated by testlab from test lab name which is "testlab"